### PR TITLE
soundwire: bus: add dev_warn() messages to track UNATTACHED devices

### PR DESCRIPTION
**Pull request by plbossart**
In rare cases, some devices seem to lose sync and never re-attach on
the bus. This seems to happen only when there are more than one device
per link, which suggests either an electrical issue, a race condition
or a state machine issue.

Add two dev_warn() messages to identify the sequence by which the
devices become UNATTACHED.

BugLink: https://github.com/thesofproject/linux/issues/3063
BugLink: https://github.com/thesofproject/linux/issues/3325
Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

File: drivers/soundwire/bus.c [hunk 1]
```diff
@@ -1749,8 +1749,11 @@ int sdw_handle_slave_status(struct sdw_bus *bus,
 			continue;
 
 		if (status[i] == SDW_SLAVE_UNATTACHED &&
-		    slave->status != SDW_SLAVE_UNATTACHED)
+		    slave->status != SDW_SLAVE_UNATTACHED) {
+			dev_warn(&slave->dev, "Slave %d state check1: UNATTACHED, status was %d\n",
```
Comment (ranj063) [inline drivers/soundwire/bus.c:1753]: I'm a bit confused about the warning @plbossart. In both cases it looks like modifying the slave status to UNATTACHED status is intentional. Why should it be a warning then?

Comment (kv2019i) [reply drivers/soundwire/bus.c:1753]: @plbossart Maybe a dev_info() instead? Would still be visible in dmesg (and seen in bug reports).

Comment (bardliao) [reply drivers/soundwire/bus.c:1753]: Slave will not be unattached for no reason. Warning is fine for me. Of cause dev_info is also fine for me.

Comment (plbossart) [reply drivers/soundwire/bus.c:1753]: the fact that the 'Slave' device is no longer synchronized and attached AFTER the enumeration is not expected. Worse, in such cases the device doesn't seem to come back and remains unattached, so I think dev_warn is exactly what we want.
dev_info() does not carry the weight of 'this is not normal'.
dev_err() would be too strong, we've seen in some cases that devices come and go, we can't flag an error in the common bus code for such a behavior.

**Review by bardliao — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]
LGTM

**Pull request merged**